Edit user registrations

sryoya opened the pull request:
# what

userの登録・編集・ログイン画面の編集　carrierwaveを導入してアカウント画像の投稿機能追加　デザインファイルを導入した上で修正
# why

user機能を整えるため　画像のアップロードをできるようにするため

sryoya committed:
carrierwaveの導入とavatarの導入

sryoya committed:
Merge branch 'master' into add_images

Conflicts:
	Gemfile
	Gemfile.lock

sryoya committed:
サインアップ画面へのデザインファイルの適用とform_forへの編集

sryoya committed:
ログイン画面へのデザインファイルの適用とform_forへの編集

sryoya committed:
devise_parameter_sanitizerでdeviseにストロングパラメーターを追加

sryoya committed:
ユーザー登録のバリデーションの設置

sryoya committed:
ストロングパラメーターへの記述ミスを修正

sryoya committed:
editファイルをform_forに修正　その他微調整

sryoya committed:
form_forにクラス指定　その他微調整

sryoya committed:
editファイルの編集とエラー修正

sryoya committed:
gitignoreでユーザー画像が開かないように設定

kuboshizuma commented on .gitignore:
```diff
@@ -7,6 +7,8 @@
 # Ignore bundler config.
 /.bundle
 
+public/uploads/tmp
```
```
public/uploads
```

がいいと思います。
今のままだと、uploadした画像がプルリクに含まれています。
(「死にたくなった」と言っているのびたとキノコの画像が…)

sryoya replied:
なるほど、、、お恥ずかしいものをお見せしましたww

sryoya committed:
gitignoreの修正

kuboshizuma commented:
画像は要らないのでプルリクから削除しましょう。

kuboshizuma commented on db/schema.rb:
```diff
@@ -11,7 +11,16 @@
+    t.string   "title",      limit: 255
+    t.string   "catch_copy", limit: 255
+    t.text     "concept",    limit: 65535
+    t.integer  "user_id",    limit: 4
```
user_idにindexを張りましょう。
indexをはるとデータ取得のパフォーマンスが上がります。
詳しくは以下を参考にしてみて下さい。
http://qiita.com/seiya1121/items/fb074d727c6f40a55f22

migrationファイル内でreferencesを使いましょう！
http://railsdoc.com/references/references

以前のプルリク範囲なのですが、schema.rbを見ていて気付きました。。

sryoya replied:
user_idにindexを貼りました！これから必要と思ったらindexを使用していこうと思います

referencesについても理解はできました！でもこれは以前のPR範囲なので、次回以降referenceを使用していくということでよろしいですかね？

kuboshizuma replied:
そうですね！
次回からははじめからreferencesを使いましょう！

sryoya replied:
了解です！！

sryoya committed:
間違ってあげた画像の削除

sryoya committed:
user_idにインデックスを貼りました

kuboshizuma commented:
LGTM！
かと思ったんだけど、conflict起こしてマージ出来ないみたいだから確認してみてほしいです！

sryoya committed:
以下のコンフリクトの修正
Merge branch 'master' into edit_user_registrations

# Conflicts:
#	app/models/user.rb
#	db/schema.rb

kuboshizuma commented:
OKです！マージしちゃって下さい！
